lerobot/common/datasets/push_dataset_to_hub/_convert_datasets.py

```diff
@@ -0,0 +1,112 @@
+import argparse
+from pathlib import Path
+
+from lerobot import available_datasets
+from lerobot.common.datasets.push_dataset_to_hub._download_raw import download_raw
+from lerobot.scripts.push_dataset_to_hub import push_dataset_to_hub
+
+SOURCE_FORMAT = {
+    "aloha": "aloha_hdf5",
+    "pusht": "pusht_zarr",
+    "xarm": "xarm_pkl",
+    "umi": "umi_zarr",
+}
+
+
+def convert_datasets(
+    local_dir: Path,
+    raw_repo_ids: list[str],
+    push_repo: str,
+    vcodec: str,
+    pix_fmt: str,
+    g: int,
+    crf: int,
+    dry_run: bool = False,
+) -> None:
+    if len(raw_repo_ids) == 1 and raw_repo_ids[0].lower() == "all":
+        raw_repo_ids = [f"cadene/{id_.split('/')[1]}_raw" for id_ in available_datasets]
+
+    for dataset_repo_id_raw in raw_repo_ids:
+        dataset_id_raw = dataset_repo_id_raw.split("/")[1]
+        dataset_id = dataset_id_raw[:-4]
+        dataset_repo_id_push = f"{push_repo}/{dataset_id}"
+        raw_format = SOURCE_FORMAT[dataset_id.split("_")[0]]
```

Comment:
Tricky. Would be nice to have a more future proof way than `SOURCE_FORMAT[dataset_id.split("_")[0]]`

Maybe by listing all the raw dataset somewhere associated with their format?

Or maybe to have a dedicated function `get_raw_format(raw_dataset_id)`?

Maybe fine for now

Reply:
Working on it to come up with a better way.

Reply:
Fixed https://github.com/huggingface/lerobot/pull/302/commits/fc3792d6ed290fa2f950221b76e104f6cd9dd5e0